### PR TITLE
remove normalization of likelihood on cl-test

### PR DESCRIPTION
**Pull request by bayonato89**

### CODE REVIEW

File: csep/core/poisson_evaluations.py [hunk 1]
```diff
@@ -486,89 +488,95 @@ def _simulate_catalog(num_events, sampling_weights, sim_fore, random_numbers=Non
...
-        observed_data (numpy.ndarray): same format as observation.
-        num_simulations: default number of simulations to use for likelihood based simulations
-        seed: used for reproducibility of the prng
-        random_numbers (numpy.ndarray): can supply an explicit list of random numbers, primarily used for software testing
-        use_observed_counts (bool): if true, will simulate catalogs using the observed events, if false will draw from poisson distrubtion
-    """
-    # set seed for the likelihood test
-    if seed is not None:
-        numpy.random.seed(seed)
+# def _poisson_likelihood_test(forecast_data, observed_data, num_simulations=1000, random_numbers=None,
```
Comment (wsavran) [inline csep/core/poisson_evaluations.py:491]: i think this comment somehow made it to the commit on accident.

File: csep/core/poisson_evaluations.py [hunk 2]
```diff
@@ -486,89 +488,95 @@ def _simulate_catalog(num_events, sampling_weights, sim_fore, random_numbers=Non
...
+                         seed=None, use_observed_counts=True, verbose=True, normalize_likelihood=False):
+"""
+Computes the likelihood-test from CSEP using an efficient simulation based approach.
+Args:
+    forecast_data (numpy.ndarray): nd array where [:, -1] are the magnitude bins.
+    observed_data (numpy.ndarray): same format as observation.
+    num_simulations: default number of simulations to use for likelihood based simulations
+    seed: used for reproducibility of the prng
+    random_numbers (numpy.ndarray): can supply an explicit list of random numbers, primarily used for software testing
+    use_observed_counts (bool): if true, will simulate catalogs using the observed events, if false will draw from poisson distribution
```
Comment (wsavran) [inline csep/core/poisson_evaluations.py:501]: these lines should be indented

**Review by wsavran — CHANGES_REQUESTED** [review]

File: csep/core/poisson_evaluations.py (hunk 2 above)
Comment (bayonato89) [reply csep/core/poisson_evaluations.py]: Indented, Bill.

**Comment by bayonato89** [comment]
The lines are properly idented now :)

**Comment by wsavran** [comment]
fixes #78

**Comment by wsavran** [comment]
the ci build/test failed because of some python tabs vs. spaces issue in the poisson_evaluations.py file. @bayonato89 i fetched the branch from your forked copied and i fixed the spacing issues in the commit. the build and tests run without error.

i believe the original implementation of this pull request didn't produce the correct normalization / conditioning for the cl test. i went through the `_poisson_likelihood_test(...)` function and greatly simplified the logic for choosing the number of events to simulate and whether we want to normalize the likelihoods. before i merge these changes, @bayonato89 @mjw98 and @khawajasim do you all mind reviewing the changes to make sure that we have the correct form implemented?

**desired behavior for normalizing and conditioning**
TEST NAME | normalize_likelihood | use_observed_counts |
| ---- | ---- | ---- | 
|M-test| True | True |
|S-test| True | True |
|CL-test| False | True |
|L-test| False | False |

**implementation from previous commit**
```python
# used for conditional-likelihood, magnitude, and spatial tests to normalize the rate-component of the forecasts.
if use_observed_counts:
    scale = n_obs / n_fore
    if normalize_likelihood:
        expected_forecast_count = numpy.sum(forecast_data)
        log_bin_expectations = numpy.log(forecast_data.ravel())
    else:    
        expected_forecast_count = int(n_obs)
        log_bin_expectations = numpy.log(forecast_data.ravel() * scale)
else:
    expected_forecast_count = numpy.sum(forecast_data)
    log_bin_expectations = numpy.log(forecast_data.ravel())

# ... code omitted ...

for idx in range(num_simulations):
    if use_observed_counts:
        if normalize_likelihood:
            num_events_to_simulate = int(n_obs)
        else:    
            num_events_to_simulate = expected_forecast_count
    else:
        num_events_to_simulate = int(numpy.random.poisson(expected_forecast_count))
```

the error appears to be in the `if normalize_likelihood` clause, because if we are normalizing the likelihood we want to scale the rates to the observed rates. additionally, the m-test and s-test are called with `normalize_likelihood=True` and l-test and cl-test are called with `normalize_likelihood=False`. i propose a fix to this along with some simplification of the logic below.

**simplified implementation**
```python
expected_forecast_count = numpy.sum(forecast_data)
log_bin_expectations = numpy.log(forecast_data.ravel())
# used for conditional-likelihood, magnitude, and spatial tests to normalize the rate-component of the forecasts
if use_observed_counts and normalize_likelihood:
    scale = n_obs / n_fore
    expected_forecast_count = int(n_obs)
    log_bin_expectations = numpy.log(forecast_data.ravel() * scale)

# ... code omitted ...

for idx in range(num_simulations):
    if use_observed_counts:
        num_events_to_simulate = int(n_obs)
    else:
        num_events_to_simulate = int(numpy.random.poisson(expected_forecast_count))
```

in this implementation, we scale the rates when normalizing the likelihood and simulating using the observed event counts, else the rates aren't scaled. we now either simulate using the number of observed events or numbers drawn from a poisson distribution. the simplified version matches our desired behavior.

**Review by wsavran — APPROVED** [review]

**Comment by wsavran** [comment]
@bayonato89 im merging this into master. nice job on finding this issue!

**Pull request merged**